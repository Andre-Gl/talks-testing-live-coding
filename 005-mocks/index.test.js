const mockStatesData = { 'TST' : 'Test' };
const mockZipLookupData = { state: 'TST' };
const mockGetStatesFn = jest.fn().mockImplementation(() => mockStatesData);

const mockZipLookupFn = jest.fn().mockImplementation(() => Promise.resolve(mockZipLookupData));

jest.mock('./data', () => ({
  getStates: mockGetStatesFn,
  zipLookup: mockZipLookupFn,
}));

const states = require('./index');

let statesAbbrList;

beforeEach(() => {
  statesAbbrList = states.getStatesAbbrList();
});

describe('getStatesAbbrList', () => {
  it('should return list of abbreviations', () => {
    expect(statesAbbrList).toMatchSnapshot();
  });
});

describe('zipLookup', () => {
  it('should resolve correct zip', () => {
    expect.assertions(1);
    return expect(states.zipLookup('94085')).resolves.toMatchSnapshot();
  });


  it('should reject on invalid zip', () => {
    expect.assertions(1);
    return expect(states.zipLookup('')).rejects.toMatchSnapshot();
  });
});
