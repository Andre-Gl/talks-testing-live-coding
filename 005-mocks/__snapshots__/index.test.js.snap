// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getStatesAbbrList should return list of abbreviations 1`] = `
Array [
  "TST",
]
`;

exports[`zipLookup should reject on invalid zip 1`] = `"Zip code is required"`;

exports[`zipLookup should resolve correct zip 1`] = `"TST"`;
